src/cvodes/cvodes.c

```diff
@@ -8669,42 +8720,44 @@ static int cvQuadSensEwtSetEE(CVodeMem cv_mem, N_Vector *yQScur, N_Vector *weigh
   pyS = cv_mem->cv_tempvQS[0];
 
   for (is=0; is<cv_mem->cv_Ns; is++) {
-    N_VScale(cv_mem->cv_pbar[is], yQScur[is], pyS);
+    SUNCheckCallLastErrNoRet(N_VScale(cv_mem->cv_pbar[is], yQScur[is], pyS));
     flag = cvQuadEwtSet(cv_mem, pyS, weightQS[is]);
     if (flag != 0) return(-1);
-    N_VScale(cv_mem->cv_pbar[is], weightQS[is], weightQS[is]);
+    SUNCheckCallLastErrNoRet(N_VScale(cv_mem->cv_pbar[is], weightQS[is], weightQS[is]));
   }
   return(0);
 }
 
 static int cvQuadSensEwtSetSS(CVodeMem cv_mem, N_Vector *yQScur, N_Vector *weightQS)
 {
+  SUNAssignSUNCTX(CV_SUNCTX);
   int is;
 
   for (is=0; is<cv_mem->cv_Ns; is++) {
-    N_VAbs(yQScur[is], cv_mem->cv_tempvQ);
-    N_VScale(cv_mem->cv_reltolQS, cv_mem->cv_tempvQ, cv_mem->cv_tempvQ);
-    N_VAddConst(cv_mem->cv_tempvQ, cv_mem->cv_SabstolQS[is], cv_mem->cv_tempvQ);
+    SUNCheckCallLastErrNoRet(N_VAbs(yQScur[is], cv_mem->cv_tempvQ));
+    SUNCheckCallLastErrNoRet(N_VScale(cv_mem->cv_reltolQS, cv_mem->cv_tempvQ, cv_mem->cv_tempvQ));
+    SUNCheckCallLastErrNoRet(N_VAddConst(cv_mem->cv_tempvQ, cv_mem->cv_SabstolQS[is], cv_mem->cv_tempvQ));
     if (cv_mem->cv_atolQSmin0[is]) {
       if (N_VMin(cv_mem->cv_tempvQ) <= ZERO) return(-1);
     }
-    N_VInv(cv_mem->cv_tempvQ, weightQS[is]);
+    SUNCheckCallLastErrNoRet(N_VInv(cv_mem->cv_tempvQ, weightQS[is]));
   }
   return(0);
 }
 
 static int cvQuadSensEwtSetSV(CVodeMem cv_mem, N_Vector *yQScur, N_Vector *weightQS)
 {
+  SUNAssignSUNCTX(CV_SUNCTX);
   int is;
 
   for (is=0; is<cv_mem->cv_Ns; is++) {
-    N_VAbs(yQScur[is], cv_mem->cv_tempvQ);
+    SUNCheckCallLastErrNoRet(N_VAbs(yQScur[is], cv_mem->cv_tempvQ));
     N_VLinearSum(cv_mem->cv_reltolQS, cv_mem->cv_tempvQ, ONE,
                  cv_mem->cv_VabstolQS[is], cv_mem->cv_tempvQ);
     if (cv_mem->cv_atolQSmin0[is]) {
       if (N_VMin(cv_mem->cv_tempvQ) <= ZERO) return(-1);
```

Comment:
As above, needs a `sunrealtype minval` declared at the top of the function, and then this should be changed to something like
```suggestion
    minval = SUNCheckCallLastErrNoRet(N_VMin(cv_mem->cv_tempvQ));
    if (minval <= ZERO) return(-1);
```